src/AbstractPeriod.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace RebelCode\Time;
+
+/**
+ * Abstract common functionality for periods of time.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractPeriod
+{
+    /**
+     * Retrieves the start time of this period.
+     *
+     * @since [*next-version*]
+     *
+     * @return TimeInterface
+     */
+    abstract protected function _getStart();
+
+    /**
+     * Retrieves the end time of this period.
+     *
+     * @since [*next-version*]
+     *
+     * @return TimeInterface
+     */
+    abstract protected function _getEnd();
+
+    /**
+     * Retrieves the duration of this period.
+     *
+     * @since [*next-version*]
+     *
+     * @return int An integer number of seconds. May be negative if the start time is larger than the end time.
```

Comment:
How can a duration be negative?

Reply:
When `start > end`. Though I have no idea why I documented this. While true, mentioning it here is just confusing. Remove?